google-cloud-spanner/src/main/java/com/google/cloud/spanner/connection/StatementParser.java

```diff
@@ -460,6 +464,12 @@ static String removeStatementHint(String sql) {
       startQueryIndex = upperCaseSql.indexOf(keyword);
       if (startQueryIndex > -1) break;
     }
+    if (startQueryIndex <= -1) {
```

Comment:
instead of this `if` statement
you can probably make changes in the previous for-loop as follows:

```
Set<String> selectAndDmlStatements = Sets.union(selectStatements, dmlStatements).immutableCopy();
for (String keyword : selectAndDmlStatements) {
```
for (String keyword :

Reply:
Please apply this suggestion, as there is no need to do this in two separate loops. Also, please update the comment on line 460 to reflect the fact that statement hints are also supported for DML statements.

Reply:
Updated both the for-loop and comment.